package/iframe/js/features.js

```diff
@@ -0,0 +1,99 @@
+var logger = require('./logger');
+
+
+// Bump this number every time you add a feature. It must match zamboni's
+// settings.APP_FEATURES_VERSION.
+var APP_FEATURES_VERSION = 8;
+
+function buildFeaturesPromises() {
+    var promises = [];
+    // The order matters - we'll push promises in the same order to
+    // generate the features signature.
+    var features = [
+        // false,  // Hardcoded boolean value.
+        // 'mozApps' in navigator,  // Dynamic boolean value.
+        // ['hardware.memory', 512],  // getFeature() with comparison.
+        // 'api.window.MozMobileNetworkInfo',  // hasFeature().
+
+        // We are only interested in a few features for now. We're already
+        // only doing this if getFeature() is present, and it was
+        // introduced in 2.0, so we know we can hardcode anything that
+        // comes with 2.0 and for which we don't need to know the exact
+        // value.
+        true, // 'getMobileIdAssertion' in window.navigator || 'api.window.Navigator.getMobileIdAssertion',
+        true, // 'manifest.precompile',
+        ['hardware.memory', 512],
+        ['hardware.memory', 1024],
+        true, // NFC
+        'acl.version', // OpenMobile ACL
+        // Don't add any more as long as bug 1172487 is not fixed, it won't
+        // work correctly.
+    ];
+
+    features.forEach(function(key) {
+        if (typeof key === 'boolean') {
```

Comment:
I prefer `key.constructor === Boolean` but the perf between this and `typeof` seems negligible